sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -52,38 +53,66 @@ def _parse_conn_str(conn_str, kwargs):
     shared_access_signature = None  # type: Optional[str]
     shared_access_signature_expiry = None # type: Optional[int]
     eventhub_name = kwargs.pop("eventhub_name", None)  # type: Optional[str]
-    for element in conn_str.split(";"):
-        key, _, value = element.partition("=")
-        if key.lower() == "endpoint":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "hostname":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "sharedaccesskeyname":
-            shared_access_key_name = value
-        elif key.lower() == "sharedaccesskey":
-            shared_access_key = value
-        elif key.lower() == "entitypath":
-            entity_path = value
-        elif key.lower() == "sharedaccesssignature":
-            shared_access_signature = value
-            try:
-                # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
-                # type: ignore
-                shared_access_signature_expiry = int(shared_access_signature.split('se=')[1].split('&')[0])
-            except (IndexError, TypeError, ValueError): # Fallback since technically expiry is optional.
-                # An arbitrary, absurdly large number, since you can't renew.
-                shared_access_signature_expiry = int(time.time() * 2)
-    if not (all((endpoint, shared_access_key_name, shared_access_key)) or all((endpoint, shared_access_signature))):
+    check_case = kwargs.pop("check_case", False)    # type: Optional[bool]
+    conn_settings = core_parse_connection_string(conn_str, case_sensitive_keys=check_case)
+    if check_case:
+        shared_access_key = conn_settings.get("SharedAccessKey")
+        shared_access_key_name = conn_settings.get("SharedAccessKeyName")
+        endpoint = conn_settings.get("Endpoint")
+        entity_path = conn_settings.get("EntityPath")
+    
+        # non case sensitive check when parsing connection string for internal use
+        for key, value in conn_settings.items():
+            # only sas check is non case sensitive for both conn str properties and internal use
+            if key.lower() == "sharedaccesssignature":
```

Comment:
if conn_settings is dict, then we don't need to iterate every key-value pair in the items?
```python
signature = conn_settings.get("sharedaccesssignature") or conn_settings.get("SharedAccessSignature")
```
but if users pass weird case like `shareDacceSSSignaturE`, then we're not able to parse it.. just thinking out loud.

Reply:
potentially we could call the core parser again and get the sas? IE)
```conn_settings = core_parse_connection_string(conn_str, case_sensitive_keys=False)
  shared_access_signature = conn_settings.get("sharedaccesssignature")
```
although that's not really any more efficient than looping through...